Login page with responsiveness and validation

Aar7 opened the pull request:
- Implemented the basic structure of the login UI
- Refactored `button.tsx` and `theme-provider.tsx` so they only export components; moved all other exports to separate files within the tree and created new directories as necessary

katemyer commented:
There appears to be some conflicts with 2 tsx files. Can you resolve these and resubmit this PR for review please?

Aar7 commented:
I'm not sure why the PR Testing Process test is failing. It's reading that I have duplicate declarations in the `theme-provider.tsx` file, and I'm not sure why.

katemyer commented:
Take a look at the import statements:

import React from "react";
import { useEffect, useState } from "react";
import { ThemeProviderContext, Theme, ThemeProviderState } from "@/contexts/theme-context";

when similar imports like:

import React, { useEffect, useState } from "react";
import { Theme, ThemeProviderContext } from "./theme-context";

were already effectively in place. This means the same names (like React, Theme, useState) were being imported or declared more than once in the same file scope.

Aar7 commented:
> Take a look at the import statements:
> 
> import React from "react"; import { useEffect, useState } from "react"; import { ThemeProviderContext, Theme, ThemeProviderState } from "@/contexts/theme-context";
> 
> when similar imports like:
> 
> import React, { useEffect, useState } from "react"; import { Theme, ThemeProviderContext } from "./theme-context";
> 
> were already effectively in place. This means the same names (like React, Theme, useState) were being imported or declared more than once in the same file scope.

I understand that. It's just unclear what I'm supposed to do to stop those errors from re-appearing when the tests run because, like I said, there are no duplicates.

There were duplicates before I resolved the merge conflicts, but there shouldn't be any more; and there aren't.

Aar7 commented:
I've fixed the errors with the theme-provider and added styles to the login-route UI.
Here's a link to my work so far: ![login-route](https://github.com/user-attachments/assets/4046b9b6-4d04-4ef6-8d1b-039a1ad5210c).

To Do next:
- Form validation
- Perfect styles to design
- Fix form-related elements to suit proper form-design
- Continue learning TailwindCSS

katemyer reviewed (approved):
beautiful!
